### DISCUSSION TITLE
Setting up authentication

### DISCUSSION THREAD

**Discussion by stdedos** [ArchiveBox/ArchiveBox#1426] [Q&A] [answered]
I am trying to setup a non-admin user for the day-to-day things (like archiving websites). 

![image](https://github.com/ArchiveBox/ArchiveBox/assets/133706/33aefdd5-b6fd-430f-b140-d128ff0e1f17)

However, that does not work 😕

![image](https://github.com/ArchiveBox/ArchiveBox/assets/133706/0348fc98-ccf5-42fb-8ff7-a791b5822b16)

Am I doing something wrong?
What access rights should I give to my user?

... also, is there a reason that the password is not read-only/"hidden"?

**Comment by pirate** [comment, marked as answer]
I'm sorry for the trouble but this feature is not available in free ArchiveBox currently. I do offer it to paying clients who hire me for consulting to set up a server for their company (among other services like security review, custom archiving scripts to work around bot-blocking, etc.).

It's essentially the only part of ArchiveBox that I can use to try and get big corporate users to pay for it, without making any of ArchiveBox closed source. Without at least *some* revenue I can't afford to keep working on archivebox full time, and the project starts to crumble.

You can however customize the UI permissions with these three options without paying anything:
https://github.com/ArchiveBox/ArchiveBox/wiki/Configuration#public_index--public_snapshots--public_add_view  
(In your case I recommend creating admin accounts for your users and trusting them not to mess things up, or setting `archivebox config --set PUBLIC_ADD_VIEW=True` so that they can add URLs without an account at all.)

I've edited our documentation just now to make this clearer: https://github.com/ArchiveBox/ArchiveBox/wiki/Setting-up-Authentication#set-up-admin-web-ui-permissions

---

As for the password field, it's exposed because it's not the actual password, it's the password hash you're editing there. It's useful to have it read/write to be able to transfer users from one server to another without exposing their plain text passwords to the admin doing the transfer, but that's an uncommon use-case so most people should not touch that field.

To change a user's password you must use `archivebox manage changepassword <username>` or the user can go to `/admin/password_change/`.

**Reply by stdedos** [reply]
I respect that. I am happy to hear that your model works, and thus you can offer us this OSS.

> In your case I recommend creating admin accounts for your users and trusting them not to mess things up

I am the same user 😅 I just hoped not to use my SUPER ADMIN password all the time (not for a "good" reason really, just cyber-security things)
